sycl/include/CL/__spirv/spirv_ops.hpp

```diff
@@ -86,6 +86,30 @@ __spirv_JointMatrixSUMadINTEL(
     __spv::__spirv_JointMatrixINTEL<T3, M, N, LC, S> *C,
     __spv::Scope::Flag Sc = __spv::Scope::Flag::Subgroup);
 
+template <typename T>
+using __spirv_wi_slice_t = T __attribute__((ext_vector_type(0xffffff)));
+
+template <typename T, std::size_t R, std::size_t C, __spv::MatrixLayout U,
+          __spv::Scope::Flag S = __spv::Scope::Flag::Subgroup>
+extern SYCL_EXTERNAL __spirv_wi_slice_t<T> &__spirv_JointMatrixGetSliceData(
+    __spv::__spirv_JointMatrixINTEL<T, R, C, U, S> *);
+
+template <typename T, std::size_t R, std::size_t C, __spv::MatrixLayout U,
+          __spv::Scope::Flag S = __spv::Scope::Flag::Subgroup>
+extern SYCL_EXTERNAL size_t __spirv_JointMatrixGetSliceLength(
+    __spv::__spirv_JointMatrixINTEL<T, R, C, U, S> *);
+
+template <typename T, std::size_t R, std::size_t C, __spv::MatrixLayout U,
+          __spv::Scope::Flag S = __spv::Scope::Flag::Subgroup>
+extern SYCL_EXTERNAL T __spirv_JointMatrixGetSliceElem(
+    __spv::__spirv_JointMatrixINTEL<T, R, C, U, S> *, size_t i);
```

Comment:
The width of `size_t` type is platform dependent I think. Typically it becomes `i32` or `i64` in LLVM IR. I would prefer a more specific type, like `uint32_t`.
Is 32 bits enough for indexing in slices, what do you think @dkhaldi?

Reply:
32 bits is enough. But size_t is widely used in other APIs. 
Specifically, what if the user is calculating this iterator using some WI id or other id which is also size_t, will that work?

Reply:
If we explicitly say that this parameter must be `uint32_t` there may be a warning about type narrowing during compilation. But now I think maybe I was wrong and we really should stick to `size_t`. On LLVM IR level we can handle it in the same way as `memcpy` (https://llvm.org/docs/LangRef.html#llvm-memcpy-intrinsic), i.e. it is overloaded type.